pyiron_base/project/generic.py

```diff
@@ -1213,6 +1213,21 @@ def queue_delete_job(self, item):
         else:
             raise EnvironmentError("copy_to: is not available in Viewermode !")
 
+    def kill_job(self, job_specifier):
+        """
+        Kills a running job via the queueing system
+
+        Args:
+            job_specifier (int, GenericJob): Provide either the job_ID or the full hamiltonian
```

Comment:
Shouldn't it be the job name?

Reply:
I mean more the variable type - now it's `int` and `GenericJob`, but if it's the job name that would be `str`. And actually does it work with the job name?

Reply:
I am not sure, actually. But this is essentially just a copy of the queue_delete_job function of the class. So if it works there, it should work here. Should we just implement this with the ability to take in job_id, job object, and job_name? So we just check for the type?

Reply:
The naming is bad - I agree, but sometimes it helps to just look for other places where this function is used: 
https://github.com/pyiron/pyiron_base/blob/master/pyiron_base/jobs/job/core.py#L420

So maybe rather than having duplicated code, the removal of the job from the queuing system can be placed in its own function which is then called in both situations.

Reply:
I have a preference for `kill_job` to `queue_delete_job` on the user-utilisation side. 

Possible to rename? I would imagine that this feature will be used by many users and `queue_delete_job` is somewhat obscure and not as clear, imho.

Reply:
I find `job.*_job` a bit.  Maybe just `job.kill()` or `job.stop()`?

Reply:
The problem is that I can see that a `kill` method already exists for the job object in `pyiron_base/pyiron_base/jobs/job/generic.py` in the `GenericJob` class.

Unfortunately, it appears that it calls `remove` which uses `remove_child` and it has a inbuilt deletion of the working directory.

What do? 

Implement a new method which doesn't remove the directories?

Or modify the kill method and extend it with mostly duplicated code? 

Or modify the remove_child and remove methods to have an if branch and a flag which toggles the deletion of the other stuff (super ugly?)

Either way I don't see a way around writing significant amounts of duplicated code or having a really ugly spaghetti solution.

The problem is that `job.kill()` calls `remove_child` and `remove`, and there is too much functionality built into these two functions that isn't consistent with the behaviour that we want.

Reply:
So; I sketched this out:

Move 
https://github.com/pyiron/pyiron_base/blob/dd27661aa00c135f0b4fcc6506a2fe8e2d9a921f/pyiron_base/jobs/job/core.py#L433-L441

into it's own little function in `pyiron_base/jobs/job/core.py` named `kill`

And then I will change the old `job.kill` method already in `generic.py` to inherit it from the `kill` method in `GenericJob` 

Any objections? I've checked that the kill method isn't currently being used anywhere. It appears to remove the folders but keep a database entry with master_id and parent_id with no job folder.